compiler/rustc_query_system/src/query/caches.rs

```diff
@@ -15,7 +15,7 @@ pub trait CacheSelector<K, V> {
 }
 
 pub trait QueryStorage: Default {
-    type Value;
+    type Value: Debug;
```

Comment:
Are bounds in this file necessary? Putting them on `QueryConfig::Value` should be enough.